core/src/main/scala/org/bitcoins/core/wallet/fee/FeeRate.scala

```diff
@@ -0,0 +1,87 @@
+package org.bitcoins.core.wallet.fee
+
+import org.bitcoins.core.currency.{CurrencyUnit, Satoshis}
+import org.bitcoins.core.protocol.transaction.Transaction
+
+/**
+  * This is meant to be an abstract type that represents different fee rates for
+  * blockchains
+  */
+sealed abstract class FeeRate {
+  def *(tx: Transaction): CurrencyUnit = calc(tx)
+  def calc(tx: Transaction): CurrencyUnit
+  def baseAmount: Double
```

Comment:
Is it safe for us to use `Double`? (what are the protections against overflow, does it lose precision or fail or?)